catalog/utilities/media_props_gen/column_parser.py

```diff
@@ -0,0 +1,125 @@
+    "upsert_strategy": "newest_non_null",
+    "base_column": None,
+}
+
+COLUMN_PROPS = COLUMN.keys()
+
+
+def file_to_ast(file_name) -> ast.Module:
+    with open(file_name) as f:
+        file_contents = f.read()
+    return ast.parse(file_contents)
+
+
+CODE = file_to_ast(COLUMNS_PATH)
+
+
+def format_python_column(
+    python_column: dict[str, any],
+    python_column_lines: dict[str, tuple[int, int]],
+) -> str:
+    """
+    Format the Python column properties dictionary to a string that can be
+    used in the markdown file.
+    """
+    col_type = python_column.pop("python_type")
+    start, end = python_column_lines[col_type]
+    python_column_string = f"[{col_type}]({COLUMNS_URL}#L{start}-L{end}) (`"
+
+    col_name = python_column.pop("name")
+    column_db_name = python_column.pop("db_name")
+    if column_db_name and col_name != column_db_name:
+        python_column_string += f'name="{col_name}", '
+
+    python_column_string += ", ".join(
+        [f"{k}={v}" for k, v in python_column.items() if v is not None]
+    )
+
+    return f"{python_column_string}`)"
+
+
+def get_python_column_types() -> dict[str, tuple[int, int]]:
+    """
+    Extract all types of columns with their line numbers for hyperlinks.
+    Sample output: `StringColumn: (3, 5)``
+    """
+    return {
+        item.name: (item.lineno, item.end_lineno)
+        for item in ast.iter_child_nodes(CODE)
+        if isinstance(item, ast.ClassDef) and item.name.endswith("Column")
+    }
+
+
+def parse_col_argument_value(item):
+    """
+    Return `attr` for Attribute value (upsert strategies), `func.id` for Call value (base_column),
+    and `value` for Constant values such as `true`.
+    We don't save the List type used for sql_args.
+    """
+    # Upsert strategy
+    if isinstance(item, ast.Attribute) and isinstance(item.value, ast.Name):
+        return item.attr
+    # Base column
+    elif isinstance(item, ast.Call) and isinstance(item.func, ast.Name):
+        return item.func.id
+    elif isinstance(item, ast.Constant):
+        return item.value
+    return item.value
+
+
+def parse_python_columns() -> dict[str, any]:
+    """
+    Parse columns.py to a dictionary with the column's `db_name` as a key,
+    and the string describing Python column as a value.
+    Example output:
+    "height": "[IntegerColumn](/link/to/column/type/definition/)" +
+    "(`name="height", nullable=True, required=False, upsert_strategy=newest_non_null`)"
+    """
+    python_column_lines = get_python_column_types()
+
+    # Extracts all the assignments of the form `column_name = <Type>Column(...)`
+    cols: list[ast.Call] = [
+        item.value
+        for item in ast.iter_child_nodes(CODE)
+        if isinstance(item, ast.Assign)
+        and isinstance(item.value, ast.Call)
+        and isinstance(item.value.func, ast.Name)
+        and item.value.func.id.endswith("Column")
+    ]
+
+    columns = {}
+    for col in cols:
+        parsed_column = copy.copy(COLUMN) | {
```

Comment:
I think defining `COLUMN` as a [`dataclass`](https://docs.python.org/3/library/dataclasses.html) or even a typed `NamedTuple` here might be a little easier 🤔 we could define defaults in the same way and then just splat the keywords into the constructor at this step. Some of this logic could even be in the `__post_init__` function rather than having to access dictionary keys repeatedly! It could help make the code a bit more maintainable down the line as well if the types/attributes are clearly defined.